src/components/Modals/RateFluctuationModal/RateFluctuationModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { useRef } from 'react';
+import { Localize } from '@deriv-com/translations';
+import { Button, Modal, Text, useDevice } from '@deriv-com/ui';
+import './RateFluctuationModal.scss';
+
+type TRateFluctuationModalProps = {
+    isModalOpen: boolean;
+    onContinue: () => void;
+    onRequestClose: () => void;
+    values: {
+        currency: string;
+        inputAmount: string;
+        localCurrency: string;
+        receivedAmount: string;
+    };
+};
+
+const RateFluctuationModal = ({ isModalOpen, onContinue, onRequestClose, values }: TRateFluctuationModalProps) => {
+    const { isMobile } = useDevice();
+    const buttonText = isMobile ? 'md' : 'sm';
+    const valuesRef = useRef<TRateFluctuationModalProps['values'] | undefined>(values);
```

Comment:
why ref?

Reply:
this is to make sure the values dont change inside of the modal, since some of the values are useState state values, and can change when the rate changes. We need to lock the values and never update it here